### ISSUE TITLE
wrong argument type RbHnswlibL2Space (expected RbHnswlibInnerProductSpace) (TypeError)

### ISSUE THREAD

**Issue by kojix2** [yoshoku/hnswlib.rb#1]
Hi!
When I ran the Usage example in README.md, I got the following error.

```
hnswlib.rb/lib/hnswlib.rb:119:in `load_index': wrong argument type RbHnswlibL2Space (expected RbHnswlibInnerProductSpace) (TypeError)
```

```
/home/kojix2/Ruby/hnswlib.rb/lib/hnswlib.rb:119:in `load_index': wrong argument type RbHnswlibL2Space (expected RbHnswlibInnerProductSpace) (TypeError)
	from /home/kojix2/Ruby/hnswlib.rb/lib/hnswlib.rb:119:in `load'
	from (irb):15:in `<main>'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /home/kojix2/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /home/kojix2/.rbenv/versions/3.0.2/bin/irb:23:in `<top (required)>'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `kernel_load'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:28:in `run'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:474:in `exec'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:30:in `dispatch'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:24:in `start'
	from /home/kojix2/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.2.22/libexec/bundle:49:in `block in <top (required)>'
	... 4 levels...
```

Is there something wrong?

**Comment by yoshoku** [comment]
@kojix2 Thank you for reporting the error. This error seems to occur only on the IRB (ver. 1.3.x) of Ruby 3.

```ruby
$ cat usage.rb
require 'hnswlib'

f = 40 # length of item vector that will be indexed.
t = Hnswlib::Index.new(n_features: f, max_item: 1000)
...

$ ruby -v
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-darwin20]
$ ruby usage.rb
[0, 340, 565, 751, 670, 601, ...
```
```ruby
$ irb -v
irb 1.2.6 (2020-09-14)
$ irb
irb(main):001:0> RUBY_VERSION
=> "2.7.4"
irb(main):002:0> require 'hnswlib'
=> true
irb(main):003:0> t = Hnswlib::Index.new(n_features: 40, max_item: 1000)
=> #<Hnswlib::Index:0x00007f97c342cdb0 @metric="l2", @index=#<Hnswlib::HierarchicalNSW:0x00007f97c342cd10 @space=#<Hnswlib::L2Space...
irb(main):004:0> t.load('test.ann')
=> nil
irb(main):005:0>
```

```ruby
$ irb -v
irb 1.3.5 (2021-04-03)
$ irb
irb(main):001:0> RUBY_VERSION
=> "3.0.2"
irb(main):002:0> require 'hnswlib'
=> true
irb(main):003:0> t = Hnswlib::Index.new(n_features: 40, max_item: 1000)
=>
#<Hnswlib::Index:0x00007ff6973945c8
...
irb(main):004:0> t.load('test.ann')
... `load_index': wrong argument type RbHnswlibL2Space (expected RbHnswlibInnerProductSpace) (TypeError)
...
```

It seems that the comparison of instance variable's klass obtained by CLASS_OF may not work correctly on IRB.
https://github.com/yoshoku/hnswlib.rb/blob/90e1326430d4b08fb6019d595a02a2c3f928ea39/ext/hnswlib/hnswlibext.hpp#L348

I would like to consider another klass comparison method.

**Comment by yoshoku** [comment]
@kojix2 I have released version 0.1.1 including a fix for this error: https://rubygems.org/gems/hnswlib/versions/0.1.1

**Comment by kojix2** [comment]
@yoshoku Thanks!
Confirmed that it works on the irb console.
